File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/parser/MovementDirection.java
package br.com.calciolari.datahub.imports.domain.parser;

/**
 * Movement direction taken only from positively identified source content.
 * Unknown stays UNKNOWN — never defaulted to OUT for revenue.
 */
public enum MovementDirection {
	OUT,
	IN,
	RETURN,
	UNKNOWN
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Parser/MovementDirection.cs
namespace Calciolari.DataHub.Imports.Domain.Parser;

/// <summary>
/// Movement direction taken only from positively identified source content.
/// Unknown stays UNKNOWN — never defaulted to OUT for revenue.
/// </summary>
public enum MovementDirection
{
    Out,
    In,
    Return,
    Unknown
}

public static class MovementDirectionNames
{
    public static string WireName(this MovementDirection direction) => direction switch
    {
        MovementDirection.Out => "OUT",
        MovementDirection.In => "IN",
        MovementDirection.Return => "RETURN",
        MovementDirection.Unknown => "UNKNOWN",
        _ => throw new ArgumentOutOfRangeException(nameof(direction), direction, null)
    };
}
